modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/spds/spds.cc

```diff
@@ -7,11 +7,279 @@
+{
+  std::vector<std::pair<int, int>> edges_to_remove;
+
+  // Compute delta for a vertex
+  auto GetVertexDelta = [&](auto v, const Graph& g)
+  {
+    double delta = 0.0;
+
+    // Get the edge weight property map
+    auto weightmap = boost::get(boost::edge_weight, g);
+
+    // Add outgoing edge weights
+    for (auto out = boost::out_edges(v, g); out.first != out.second; ++out.first)
+      delta += weightmap[*out.first];
+
+    // Subtract incoming edge weights
+    for (auto in = boost::in_edges(v, g); in.first != in.second; ++in.first)
+      delta -= weightmap[*in.first];
+
+    return delta;
+  };
+
+  std::vector<size_t> s1, s2, s;
+  bool done = false;
+  while (!done)
+  {
+    done = true;
+
+    // Remove sinks (vertices with out-degree 0)
+    bool found_all_sinks = false;
+    while (not found_all_sinks)
+    {
+      found_all_sinks = true;
+      for (auto v : scc_vertices)
+      {
+        if (not g[v].active)
+          continue;
+
+        if (boost::out_degree(v, g) == 0)
+        {
+          g[v].active = false;
+          s2.push_back(v);
+          found_all_sinks = false;
+        }
+      }
+    }
+
+    // Remove sources (vertices with in-degree 0)
+    bool found_all_sources = false;
+    while (not found_all_sources)
+    {
+      found_all_sources = true;
+      for (auto v : scc_vertices)
+      {
+        if (not g[v].active)
+          continue;
+
+        if (boost::in_degree(v, g) == 0)
+        {
+          g[v].active = false;
+          s1.push_back(v);
+        }
+      }
+    }
+
+    double max_delta = -100.0;
+    Graph::vertex_descriptor max_delta_vertex = boost::graph_traits<Graph>::null_vertex();
+
+    for (auto v : scc_vertices)
+    {
+      if (not g[v].active)
+        continue;
+
+      double delta = GetVertexDelta(v, g);
+      if (delta > max_delta)
+      {
+        max_delta = delta;
+        max_delta_vertex = v;
+      }
+    }
+
+    if (max_delta_vertex != boost::graph_traits<Graph>::null_vertex())
+    {
+      g[max_delta_vertex].active = false;
+      s1.push_back(max_delta_vertex);
+    }
+
+    for (auto v : scc_vertices)
+    {
+      if (g[v].active)
+      {
+        done = false;
+        break;
+      }
+    }
+  }
+
+  s.reserve(s1.size() + s2.size());
+  for (size_t u : s1)
+    s.push_back(u);
+  for (size_t u : s2)
+    s.push_back(u);
+
+  for (size_t u : scc_vertices)
+  {
+    // Loop through outgoing edges
+    for (auto ei = boost::out_edges(u, g).first; ei != boost::out_edges(u, g).second; ++ei)
+    {
+      size_t v = boost::target(*ei, g);
+
+      // Check if v appears earlier in the sequence (i.e., is before u in s)
+      auto pos_u = std::find(s.begin(), s.end(), u);
+      auto pos_v = std::find(s.begin(), s.end(), v);
+      if (pos_v < pos_u)
+        edges_to_remove.emplace_back(u, v);
+    }
+  }
+
+  return edges_to_remove;
+}
+
+// Find bi-, tri-, and n-connected SCCs using Tarjan's algorithm
```

Comment:
This comment is associated with `FindSCCs` (sort of). And this method has its own doxygen comment, so maybe just remove this. Looks like a leftover from moving code around.

Reply:
Removed.